### PR TITLE
add new keyboard rad

### PR DESCRIPTION
**Pull request by anubhavd7**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
> You do not have permissions to merge. You need to follow the PR checklist and wait for us to review and merge.

Also, please **_stop_** closing and opening new PRs. Push additional changes to the same branch.

File: keyboards/rad/Makefile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+RAD
```
Comment (drashna) [inline keyboards/rad/Makefile:1]: This file looks like it's supposed to be a readme, and would need to be renamed to `readme.md`

File: keyboards/rad/config.h [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/rad/config.h:60]:
```suggestion
```

File: keyboards/rad/config.h [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x6060
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    qmkbuilder
+#define PRODUCT         keyboard
+#define DESCRIPTION     Keyboard
```
Comment (drashna) [inline keyboards/rad/config.h:25]:
```suggestion
```

File: keyboards/rad/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (drashna) [inline keyboards/rad/keymaps/default/keymap.c:41]:
```suggestion
#include QMK_KEYBOARD_H

const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
[0] = LAYOUT(
		KC_SPC, KC_SPC, 
		KC_SPC, KC_SPC, KC_SPC, 
		KC_SPC, KC_SPC, KC_SPC, 
		KC_SPC, KC_SPC, KC_SPC)
}
```

File: keyboards/rad/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+			    ____ , ____, 
+                ____ , ____, ____,
+                ____ , ____, ____,
+				____ , ____, ____
+		)
+}
+
+
+
+
```
Comment (drashna) [inline keyboards/rad/keymaps/via/keymap.c:46]:
```suggestion
#define ____ KC_TRNS
const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
[0] = LAYOUT(
		KC_SPC, KC_SPC, 
		KC_SPC, KC_SPC, KC_SPC, 
		KC_SPC, KC_SPC, KC_SPC, 
		KC_SPC, KC_SPC, KC_SPC),
		[1] = LAYOUT(
			    ____ , ____, 
                ____ , ____, ____,
                ____ , ____, ____,
				____ , ____, ____
		),
		[2] = LAYOUT(
			    ____ , ____, 
                ____ , ____, ____,
                ____ , ____, ____,
				____ , ____, ____
		),
		[3] = LAYOUT(
			    ____ , ____, 
                ____ , ____, ____,
                ____ , ____, ____,
				____ , ____, ____
		)
}
       
```

File: keyboards/rad/keymaps/via/vial.json [hunk 6]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (drashna) [inline keyboards/rad/keymaps/via/vial.json:1]: This appears to be a VIAL json. 
https://docs.qmk.fm/#/pr_checklist?id=keyboard-prs

> Do not include VIAL json files in the PR. These do not belong in the QMK repository as they are not used by QMK firmware – they belong in the VIAL

File: keyboards/rad/rad.h [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+
+#include "quantum.h"
+
+#define LAYOUT_horizontal( \
```
Comment (drashna) [inline keyboards/rad/rad.h:20]:
```suggestion
#define LAYOUT( \
```

File: keyboards/rad/info.json [hunk 8]
```diff
@@ -0,0 +1,58 @@
+{
+  "keyboard_name": "rad",
+  "maintainer": "anubhavd7",
+  "usb": {
+    "vid": "0xFEED",
+    "pid": "0x6060",
+},
+  "layouts": {
+    "LAYOUT_horizontal": {
```
Comment (drashna) [inline keyboards/rad/info.json:9]:
```suggestion
    "LAYOUT": {
```

File: keyboards/rad/rules.mk [hunk 9]
```diff
@@ -0,0 +1,11 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+EXTRAKEY_ENABLE 	= yes
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+CONSOLE_ENABLE		= yes
+COMMAND_ENABLE		= yes
```
Comment (drashna) [inline keyboards/rad/rules.mk:10]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/rad/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+	  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+	  * GNU General Public License for more details. 
+	  * 
+	  * You should have received a copy of the GNU General Public License 
+	  * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+	  */ 
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] = LAYOUT_horizontal(
```
Comment (zvecr) [inline keyboards/rad/keymaps/default/keymap.c:17]:
```suggestion
[0] = LAYOUT(
```

File: keyboards/rad/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (zvecr) [inline keyboards/rad/keymaps/default/keymap.c:36]:
```suggestion
```

File: keyboards/rad/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+	  * but WITHOUT ANY WARRANTY; without even the implied warranty of 
+	  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+	  * GNU General Public License for more details. 
+	  * 
+	  * You should have received a copy of the GNU General Public License 
+	  * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+	  */ 
+#include QMK_KEYBOARD_H
+
+#define ____ KC_TRNS
```
Comment (zvecr) [inline keyboards/rad/keymaps/via/keymap.c:17]:
```suggestion
```

File: keyboards/rad/keymaps/via/rules.mk [hunk 13]
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE 		= yes
+VIAL_ENABLE		= yes
```
Comment (zvecr) [inline keyboards/rad/keymaps/via/rules.mk:2]:
```suggestion
```

File: keyboards/rad/config.h [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+#define UNUSED_PINS
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+
+
+
```
Comment (zvecr) [inline keyboards/rad/config.h:35]:
```suggestion
```

File: keyboards/rad/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+	  */ 
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] = LAYOUT_horizontal(
+		KC_SPC, KC_SPC, 
+		KC_SPC, KC_SPC, KC_SPC, 
+		KC_SPC, KC_SPC, KC_SPC, 
+		KC_SPC, KC_SPC, KC_SPC)
+}
```
Comment (zvecr) [inline keyboards/rad/keymaps/default/keymap.c:22]:
```suggestion
};
```

File: keyboards/rad/keymaps/via/keymap.c [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+		[3] = LAYOUT(
+			    ____ , ____, 
+                ____ , ____, ____,
+                ____ , ____, ____,
+				____ , ____, ____
+		)
+}
+
+
+
```
Comment (zvecr) [inline keyboards/rad/keymaps/via/keymap.c:45]:
```suggestion
		[0] = LAYOUT(
			    KC_SPC, KC_SPC, 
			    KC_SPC, KC_SPC, KC_SPC, 
			    KC_SPC, KC_SPC, KC_SPC, 
			    KC_SPC, KC_SPC, KC_SPC),
		[1] = LAYOUT(
			    _______ , _______, 
                _______ , _______, _______,
                _______ , _______, _______,
				_______ , _______, _______
		),
		[2] = LAYOUT(
			    _______ , _______, 
                _______ , _______, _______,
                _______ , _______, _______,
				_______ , _______, _______
		),
		[3] = LAYOUT(
			    _______ , _______, 
                _______ , _______, _______,
                _______ , _______, _______,
				_______ , _______, _______
		)
};
```

File: keyboards/rad/readme.md [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+make rad:default
+Flashing example for this keyboard (after setting up the bootloadHID flashing environment)
+make rad:default:flash
+I have two example keymaps available, both the default and the encoderlayers keymap which demonstrates using different encoder functions based on the layer chosen.
+See the build environment setup and the make instructions for more information. Brand new to QMK? Start with our Complete Newbs Guide.
+
+
+
+
+
```
Comment (zvecr) [inline keyboards/rad/readme.md:16]:
```suggestion
# rad

The Rad Macro Pad is a USB-C,  Macro Pad with 11 keys powered by QMK

* Keyboard Maintainer: Anubhav Dhiman 
* Hardware Supported: ATmega32U4
* Hardware Availability: Interest Check

Make example for this keyboard (after setting up your build environment):

    make rad:default

Flashing example for this keyboard:

    make rad:default:flash

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by zvecr** [comment]
As well as the keyboard passing a basic compile, it seems line endings on at least the readme are incorrect.

File: keyboards/rad/readme.md [hunk 18]
```diff
@@ -0,0 +1,20 @@
+# rad
+
+The Rad Macro Pad is a USB-C,  Macro Pad with 11 keys powered by QMK
+
+* Keyboard Maintainer: Anubhav Dhiman 
+* Hardware Supported: ATmega32U4
+* Hardware Availability: Interest Check
+* 
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/rad/readme.md:9]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/rad/readme.md [hunk 19]
```diff
@@ -0,0 +1,20 @@
+# rad
+
+The Rad Macro Pad is a USB-C,  Macro Pad with 11 keys powered by QMK
+
+* Keyboard Maintainer: Anubhav Dhiman 
+* Hardware Supported: ATmega32U4
+* Hardware Availability: Interest Check
+*
```
Comment (drashna) [inline keyboards/rad/readme.md:8]:
```suggestion


```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/rad/info.json [hunk 20]
```diff
@@ -0,0 +1,58 @@
+{
+  "keyboard_name": "rad",
+  "maintainer": "anubhavd7",
```
Comment (fauxpark) [inline keyboards/rad/info.json:3]:
```suggestion
  "manufacturer": "anubhavd7",
  "maintainer": "anubhavd7",
```

**Comment by zvecr** [comment]
> yes sir i have checked it, it works and even i have made the changes as per asked by you

It does not.

```
tmk_core/protocol/usb_descriptor.c:437:35: error: 'DEVICE_VER' undeclared here (not in a function)
  437 |     .ReleaseNumber              = DEVICE_VER,
```

Note: PRs are reviewed in peoples free time, pinging multiple times will not make the process happen any faster

**Comment by fauxpark** [comment]
> Note: PRs are reviewed in peoples free time, pinging multiple times will not make the process happen any faster

**Pull request merged**